src/Types.hs

```diff
@@ -551,20 +579,20 @@ type instance IdOf User             = AUID User
 type instance IdOf Idea             = AUID Idea
 type instance IdOf Topic            = AUID Topic
 type instance IdOf Delegation       = AUID Delegation
-type instance IdOf Comment          = AUID Comment
--- ^ A more precise identifier would be:
---      IdOf Comment = (AUID Idea, NonEmpty (AUID Comment))
--- This would be the full path from AulaData down to the comment
--- which could be a reply to reply...
-type instance IdOf CommentVote      = AUID User
-type instance IdOf IdeaVote         = AUID User
-type instance IdOf IdeaLike         = AUID User
--- ^ If we were to have more precise comment identifiers these 3 could
--- become: (IdOf Comment, AUID User)
+type instance IdOf Comment          = CommentId
+type instance IdOf CommentVote      = CommentVoteId
+type instance IdOf IdeaVote         = IdeaVoteLikeId
+type instance IdOf IdeaLike         = IdeaVoteLikeId
 type instance IdOf IdeaVoteResult   = AUID IdeaVoteResult
 type instance IdOf IdeaJuryResult   = AUID IdeaJuryResult
 
-type AMap a = Map (IdOf a) a
+type family   AUIDOf a
```

Comment:
As `CommentId` and `IdeaVoteLikeId`, `CommentVoteId` are composite keys for some information in the database. `AUIDOf` should be called `AKey` or something similar. I am not sure what the `A` preffix stands for in the `AUID`.